### PR TITLE
Feat: Map Node Stats (Version, Operating System etc) to each Node in Feed

### PR DESCRIPTION
**Pull request by jsdw**
This builds on #587 and addresses a couple of nits:

- Hide new oclumns by default; they take up too much space
- Default to showing '-' in new columns when no data is available.

When I tested this locally, I saw no data for a bunch of the new columns and need to understand why that is before merging.

### CODE REVIEW

**Comment by CyndieKamau** [comment]
Hey @jsdw I can answer that for you.

So when setting up a local dev node, depending on the type of OS for your machine then the output is different.

For instance, in macOS you can only see around 3 data columns, but for Linux all the columns are filled. This is something I also noted down and informed @bLd75 about.

So this setup was configured to collect more information from a linux environment than macos, so the telemetry server will get more details from linux than macos. 

So to confirm everything was working we had to test on a linux environment.

**Comment by MrishoLukamba** [comment]
> This builds on #587 and addresses a couple of nits:
> 
> * Hide new oclumns by default; they take up too much space
> * Default to showing '-' in new columns when no data is available.
> 
> When I tested this locally, I saw no data for a bunch of the new columns and need to understand why that is before merging.

We did it in a way that you have to toggle to actually display new columns. So by default they are hidden

**Comment by jsdw** [comment]
> We did it in a way that you have to toggle to actually display new columns. So by default they are hidden

If you have already edited the telemetry settings, they are saved in localStorage and so it might be that that was the case for you, and so adding new things were hidden.

When I tested it, I had no settings saved and so they all showed up by default (see the commit here where I set them to false)

**Comment by jsdw** [comment]
> So to confirm everything was working we had to test on a linux environment.

Ah, thankyou for this! I'll point a linux based node to it when I'm back home and have my linux laptop to hand :)

**Comment by MrishoLukamba** [comment]
> > So to confirm everything was working we had to test on a linux environment.
> 
> Ah, thankyou for this! I'll point a linux based node to it when I'm back home and have my linux laptop to hand :)

any updates on this, as it is ready to be merged

**Comment by jsdw** [comment]
Sorry; this fell off my radar again! We'll aim to get it reviewed and, if all is well testing a linux node pointing to it, merged, in the coming days!

File: frontend/src/components/List/Column/LinuxDistroColumn.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//Column for specifying type of distro each node is running (Linux Distribution)
```
Comment (niklasad1) [inline frontend/src/components/List/Column/LinuxDistroColumn.tsx:17]:
```suggestion
// Column for specifying type of distro each node is running (Linux Distribution)
```

**Review by niklasad1 — APPROVED** [review]

File: frontend/src/components/List/Column/VersionColumn.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import * as React from 'react';
+import { ColumnProps } from './';
+import { Node } from '../../../state';
+import icon from '../../../icons/file-binary.svg';
+
+export class VersionColumn extends React.Component<ColumnProps> {
+  public static readonly label = 'version';
+  public static readonly icon = icon;
+  public static readonly width = 154;
+  public static readonly setting = 'version';
```
Comment (pkhry) [inline frontend/src/components/List/Column/VersionColumn.tsx:28]: Not used anywhere?

File: frontend/src/components/List/Column/index.ts [hunk 3]
```diff
@@ -33,3 +33,12 @@ export * from './BlockTimeColumn';
 export * from './BlockPropagationColumn';
 export * from './LastBlockColumn';
 export * from './UptimeColumn';
+export * from './CpuArchitectureColumn'; //extra columns added
```
Comment (pkhry) [inline frontend/src/components/List/Column/index.ts:36]: Dq: but given new column follow the same structure and same pattern, can't they be factored out into one?

File: frontend/src/state.ts [hunk 4]
```diff
@@ -102,7 +113,18 @@ export class Node {
...
+    const [
+      name,
+      implementation,
+      version,
+      validator,
+      networkId,
+      target_os,
+      target_arch,
+      target_env,
+      customNullField,
```
Comment (pkhry) [inline frontend/src/state.ts:125]: OK, so this is the reason for undefined earlier?

File: frontend/src/components/List/Column/index.ts (hunk 3 above)
Comment (jsdw) [reply frontend/src/components/List/Column/index.ts:36]: Yeah I'm sure we could simplify this stuff a bunch!

Comment (jsdw) [reply frontend/src/components/List/Column/index.ts:36]: Agreed

**Review by pkhry — APPROVED** [review]

**Comment by MrishoLukamba** [comment]
wen merge? @jsdw @niklasad1 @pkhry

**Comment by jsdw** [comment]
Merged :)

**Comment by CyndieKamau** [comment]
Yay! Thank you @jsdw @niklasad1 @pkhry for the reviews! :)

**Comment by CyndieKamau** [comment]
@jsdw I think we can safely close the first PR yes? It's still open

**Comment by jsdw** [comment]
Yup; FYI I have just deployed this to live, so the new columns are now available on telemetry.polkadot.io.

Thanks for the contribution guys; nice work!

**Comment by bLd75** [comment]
Great job here, very useful infos added on nodes!
Just a remark: when adding several fields, names get truncated a lot in the list. 
It seems the table was designed for a max size and name is the only adjustable column, it would be great if it was scrollable horizontally and keep full columns length.
image

**Comment by jsdw** [comment]
Yup, and scrolling to the right is naff too; I added #592 for this!

**Pull request merged**